cmd/notation/inspect.go

```diff
@@ -0,0 +1,327 @@
+		ref.Reference = manifestDesc.Digest.String()
+	}
+
+	if opts.outputFormat == cmd.OutputPlaintext {
+		fmt.Println("Inspecting all signatures for signed artifact")
+		fmt.Println(ref.String())
+	}
+
+	output := inspectOutput{MediaType: manifestDesc.MediaType, Signatures: []signatureOutput{}}
+	skippedSignatures := false
+	err = sigRepo.ListSignatures(ctx, manifestDesc, func(signatureManifests []ocispec.Descriptor) error {
+		for _, sigManifestDesc := range signatureManifests {
+			sigBlob, sigDesc, err := sigRepo.FetchSignatureBlob(ctx, sigManifestDesc)
+			if err != nil {
+				logSkippedSignature(sigDesc, err)
+				skippedSignatures = true
+				continue
+			}
+
+			sigEnvelope, err := signature.ParseEnvelope(sigDesc.MediaType, sigBlob)
+			envelopeContent, err := sigEnvelope.Content()
+			if err != nil {
+				logSkippedSignature(sigDesc, err)
+				skippedSignatures = true
+				continue
+			}
+
+			signedArtifactDesc, err := envelope.DescriptorFromSignaturePayload(&envelopeContent.Payload)
+			if err != nil {
+				logSkippedSignature(sigDesc, err)
+				skippedSignatures = true
+				continue
+			}
+
+			signatureAlgorithm, err := proto.EncodeSigningAlgorithm(envelopeContent.SignerInfo.SignatureAlgorithm)
+			if err != nil {
+				logSkippedSignature(sigDesc, err)
+				skippedSignatures = true
+				continue
+			}
+
+			sig := signatureOutput{
+				MediaType:             sigDesc.MediaType,
+				Digest:                sigDesc.Digest.String(),
+				SignatureAlgorithm:    string(signatureAlgorithm),
+				SignedAttributes:      getSignedAttributes(envelopeContent),
+				UserDefinedAttributes: signedArtifactDesc.Annotations,
+				UnsignedAttributes:    getUnsignedAttributes(envelopeContent),
+				Certificates:          getCertificates(envelopeContent),
+				SignedArtifact: artifact{
+					MediaType: signedArtifactDesc.MediaType,
+					Digest:    signedArtifactDesc.Digest.String(),
+					Size:      signedArtifactDesc.Size,
+				},
+			}
+
+			output.Signatures = append(output.Signatures, sig)
+		}
+		return nil
+	})
+
+	if err != nil {
+		return err
+	}
+
+	err = printOutput(opts.outputFormat, output)
+	if err != nil {
+		return err
+	}
+
+	if skippedSignatures {
+		return errors.New("at least one signature was skipped and not displayed")
+	}
+
+	return nil
+}
+
+func logSkippedSignature(sigDesc ocispec.Descriptor, err error) {
+	fmt.Fprintf(os.Stderr, "Warning: Skipping signature %s because of error: %v\n", sigDesc.Digest.String(), err)
+}
+
+func getSignedAttributes(envContent *signature.EnvelopeContent) map[string]string {
+	signedAttributes := map[string]string{}
+
+	signedAttributes["signingScheme"] = string(envContent.SignerInfo.SignedAttributes.SigningScheme)
+	signedAttributes["signingTime"] = envContent.SignerInfo.SignedAttributes.SigningTime.String()
+	signedAttributes["expiry"] = envContent.SignerInfo.SignedAttributes.Expiry.String()
```

Comment:
According to spec, those datetime should be in the RFC3339 form for the JSON output

Reply:
Please note that the current format does not match the format for the tree view as in the spec.